package-spotlight-class-metadata.md

```diff
@@ -0,0 +1,210 @@
+your codebase for any `#[Alias]` attributes and generates a mapping file that enables
+superfast lookups. The mapping is pure PHP arrays, so no reflection or scanning is
+needed at runtime. But, this does mean you need to run `composer dump-autoload`
+whenever you *add or change an alias*.
+***
+
+## Aliases at Runtime
+
+You can get the alias for a class or object like so:
+
+```php
+use Zenstruck\Alias;
+
+Alias::for(ProductCategory::class); // "product-category"
+Alias::for(new ProductCategory()); // "product-category"
+```
+
+In our auditing example, you'd use this when saving activity to the database:
+
+```php
+function logActivity(object $entity, int $entityId, string $action): void
+{
+    if (!$alias = Alias::for($entity)) {
+        return; // no alias 
+    }
+
+    $this->saveToDatabase($alias, $entityId, $action);
+}
+```
+
+***NOTE
+`Alias::for()` will return `null` if no alias is found for the given class or object.
+***
+
+To go *the other way* - from alias to class name - you can do:
+
+```php
+Alias::classFor('product-category'); // "App\Entity\ProductCategory"
+```
+
+After querying our database for activity, we can use this to get the class name
+from the stored alias:
+
+```php
+class Activity
+{
+    // ...
+
+    public function getEntityClass(): ?string
+    {
+        return Alias::classFor($this->entityAlias);
+    }
+
+    // ...
+}
+```
+
+***NOTE
+`Alias::classFor()` will return `null` if no class is found for the given alias.
+***
+
+## Alternative Alias Configuration
+
+What about classes you don't control, like vendor classes? No problem! You can configure
+these in your app's `composer.json` file:
+
+```json
+{
+    "extra": {
+        "class-metadata": {
+            "map": {
+                "Galactic\\Fleet\\Hyperdrive": "hyperdrive"
+            }
+        }
+    }
+}
+```
+
+## Additional Metadata
+
+Say we don't want *all* entities to be auditable - they need to be *marked* as such.
+We have a couple options:
+
+1. Use a marker interface (an interface without methods), e.g. `AuditableInterface`, and
+   have auditable entities implement it.
+2. Create a custom class attribute, e.g. `#[Auditable]`, and add it to auditable entities.
+
+These both work, but this package provides a third option: the `#[Metadata]` attribute:
+
+```php
+use Zenstruck\Metadata;
+
+#[Metadata('auditable', true)]
+class ProductCategory
+{
+    // ...
+}
+```
+
+***IMPORTANT
+Metadata is also picked up by the Composer plugin during the autoload dump process. Be sure to run:
+
+```terminal
+composer dump-autoload
+```
+
+whenever you add or change metadata.
+***
+
+Now, before logging activity, we first check if the entity is auditable:
+
+```php
+use Zenstruck\Metadata;
+
+function onEntityChange(object $entity, int $entityId, string $action): void
+{
+    if (!Metadata::get($entity, 'auditable')) {
+        return; // not auditable
+    }
+
+    $this->logActivity($entity, $entityId, $action);
+}
```

Comment:
Same here, IMO it might not be clear we're talking about methods for that `Activity` class. I would wrap it in a class